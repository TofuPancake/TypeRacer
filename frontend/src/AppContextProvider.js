import React, { useState, useEffect } from 'react';
import { auth } from './firebase';
import themeMusic from './sound/theme_music.mp3';

const AppContext = React.createContext();

function AppContextProvider({ children }) {
    const [firebaseUserIdToken, setFirebaseUserIdToken] = useState(null);
    const [username, setUsername] = useState('');
    const [lobbyid, setLobbyid] = useState('');
    const [lobby, setLobby] = useState(null);
    const audio = new Audio(themeMusic);
    audio.loop = true;
    audio.volume = 0.1;

    useEffect(() => {
        auth.onIdTokenChanged(async (auth) => {
            if (auth) {
                const token = await auth.getIdToken();
                setFirebaseUserIdToken(token);
            } else {
                setFirebaseUserIdToken(null);
            }
        });
    }, []);

    const context = {
        firebaseUserIdToken,
        username,
        setUsername,
        lobbyid,
        setLobbyid,
        lobby,
        setLobby,
        audio
    }

    return (
        <AppContext.Provider value={context}>
            {children}
        </AppContext.Provider>
    );
}

export {
    AppContext,
    AppContextProvider,
}
