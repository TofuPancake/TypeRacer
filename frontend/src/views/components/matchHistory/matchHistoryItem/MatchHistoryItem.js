import { Collapse } from 'material-ui-next';
import { useState } from 'react';
import { makeStyles } from '@material-ui/core';
import styles from './MatchHistoryItem.module.css';
import MatchPlayersList from './matchPlayersList/MatchPlayersList';

const useStyles = makeStyles(() => ({
    collapse: {
        width: 'inherit'
    }
}))

export default function MatchHistoryItem({match}){
    const [open, setOpen] = useState(false);
    const classes = useStyles();

    function handleMouseEnter() {
        setOpen(true);
    };

    function handleMouseLeave() {
        setOpen(false);
    };

    return(
        <Collapse className={classes.collapse} in={open} collapsedSize={40}>      
            <div className={styles.matchHistoryContainer} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}> 
                <div className={styles.matchHistoryTitle}>
                    <h1 className={styles.placeText}>#{match.place} </h1><h1 className={styles.wpmText}> WPM: {match.wpm}</h1> 
                    <h2 className={styles.dateText}>{match.date}</h2>
                </div>
                <div  className={styles.gameInfo}>
                    <div  className={styles.textTypedContainer}>
                        <p>Typed: </p><p className={styles.textTyped}>{match.textTyped} </p>
                    </div>
                    <MatchPlayersList players={match.otherPlayers}/>
                </div>
            </div>
        </Collapse>
    );
}
