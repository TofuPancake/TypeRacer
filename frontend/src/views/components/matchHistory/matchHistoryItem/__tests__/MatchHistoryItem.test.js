import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AppContextProvider } from '../../../../../AppContextProvider';

import MatchHistoryItem from '../MatchHistoryItem';

test('match history item renders correctly', () => {
  const shallowRenderer = new ShallowRenderer();
  const snapshotComponent = shallowRenderer.render(
    <AppContextProvider>
      <MatchHistoryItem />
    </AppContextProvider>);
  expect(snapshotComponent).toMatchSnapshot();
});