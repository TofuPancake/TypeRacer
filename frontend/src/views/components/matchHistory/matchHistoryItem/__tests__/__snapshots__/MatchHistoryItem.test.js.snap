// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match history item renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "audio": <audio
        loop=""
        preload="auto"
        src="mock file"
      />,
      "firebaseUserIdToken": null,
      "lobby": null,
      "lobbyid": "",
      "setLobby": [Function],
      "setLobbyid": [Function],
      "setUsername": [Function],
      "username": "",
    }
  }
>
  <MatchHistoryItem />
</Context.Provider>
`;
